### ISSUE TITLE
Failed files not getting aborted with IoTHubClient_LL_UploadMultipleBlocksToBlobEx

### ISSUE THREAD

**Issue by trillykins** [Azure/azure-iot-sdk-c#573]
I have an issue with uploading a lot of files. It seems that failed files aren't aborted as I understood they would with the following code using the multiple blocks function.

```
static IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_RESULT getDataCallback(IOTHUB_CLIENT_FILE_UPLOAD_RESULT result, unsigned char const** data, size_t* size, void* context)
{
    if (result == FILE_UPLOAD_OK) {
        if (data != NULL && size != NULL) {
            if (block_count < totalSize) {
                *data = (const unsigned char*)buffer;
                *size = totalSize; //strlen(buffer);
                block_count += totalSize;
            } else {
                *data = NULL;
                *size = 0;
            }
        } else {
            // The last call to this callback is to indicate the result of uploading the previous data block provided.
            // Note: In this last call, data and size pointers are NULL.
        }
    } else {
        // (void)print_param("Received unexpected result %s", ENUM_TO_STRING(IOTHUB_CLIENT_FILE_UPLOAD_RESULT, result));
        print("Aborting file upload");
        FILE *fp = fopen(ERROR_LOG, "a+");
        if(fp) {
            fprintf(fp, "Trying to abort file upload\n");
            fclose(fp);
        }
        return IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_ABORT;
    }

    return IOTHUB_CLIENT_FILE_UPLOAD_GET_DATA_OK;
}
```

Can someone tell me if I am doing anything wrong when aborting a file upload?

**Comment by jspaith** [comment]
First, do you need to upload a lot of (presumably small?) files?   If so you should research the quota limitations of Azure IoT backend since you can trigger these if you're not careful.  There are scenarios where it makes sense - like over unreliable networks - but if you can potentially compress / merge multiple streams together (from the device=>Hub perspective, though I don't know your network backend beyond that obviously) there can be advantages.

For this particular issue, it looks like you're only trying to abort when `FILE_UPLOAD_OK` isn't set.  From a quick review of code, the only place I'm seeing the IoT stack set an error is at the very end of our upload routine [here](https://github.com/Azure/azure-iot-sdk-c/blob/a5ef2d9b82a103a52b8517fb1675f842e547e008/iothub_client/src/iothub_client_ll_uploadtoblob.c#L1054).  By this point, we throw away your return code since there's nothing else we're doing.

Does this help?  What are you expecting to see?  If not, try setting the device option (your API may or may not have the _LL_, and maybe IoTHubCore_*Upload depending on how you create the handle) for extra logging.

```C
        bool traceOn = true;
        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_LOG_TRACE, &traceOn);
```

**Comment by trillykins** [comment]
Yes, I need to upload a sequence of images, so I should, as you suggest, look at the quota limitations and adjust accordingly. 

In hindsight, I think the problem has been caused by the application crashing (debugging a segfault) and or being terminated by the user. I suspect  there isn't something that can be done in those instances to abort those file uploads? It seems like failed uploads are aborted as intended otherwise.

I have the extra logging enabled, but I forgot to include the stack trace in the issue request (and of course I don't have access to it at the moment). It said that the number of active uploads were exceeded. I use the _LL_ variant. 

But, I think the issue has been the application crashing or being terminated while uploading repeatedly during bug testing. So I think this issue can be closed. Thank you for the quick reply!
